ui/src/pages/backoffice/Loan/LoanSearch/SearchDateRange.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import { Card } from 'semantic-ui-react';
+import { DatePicker } from '@components';
+import { withState } from 'react-searchkit';
+import _isEmpty from 'lodash/isEmpty';
+
+class _SearchDateRange extends Component {
+  getCurrentDates() {
+    const { filters } = this.props.currentQueryState;
+    let loans_from_date = '';
+    let loans_to_date = '';
+
+    filters.forEach(([name, value]) => {
+      if (name === 'loans_from_date') loans_from_date = value;
+      if (name === 'loans_to_date') loans_to_date = value;
+    });
+    return [loans_from_date, loans_to_date];
+  }
+
+  onDateChange = newFilter => {
+    const [name, value] = newFilter;
+    let result = newFilter;
+
+    if (_isEmpty(value)) {
+      this.props.updateQueryState({ filters: result });
+      return;
+    }
+
+    const { filters } = this.props.currentQueryState;
+    const existingFilters = filters.filter(filter => filter[0] === name);
+
+    if (!_isEmpty(existingFilters)) {
+      result = existingFilters.push(newFilter);
```

Comment:
@topless This here is wrong as the result is assigned with the number of elements in the `existingFilters` array.
```
a = [1]
b = a.push(2)
Console:  a = [1,2]
Console: b = 2
```